infrastructure/lib/stacks/gitHubWorkflowMonitorAlarms.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+import {Duration, Stack} from "aws-cdk-lib";
+import {Construct} from "constructs";
+import {Alarm, ComparisonOperator, Metric, TreatMissingData} from "aws-cdk-lib/aws-cloudwatch";
+
+export interface AlarmProps {
+    readonly namespace: string;
+    readonly metricName: string;
+    readonly workflows: string[];
+}
+
+export class GitHubWorkflowMonitorAlarms extends Stack {
+    readonly workflowAlarmsArn: string[] = [];
+    readonly metricName: string
```

Comment:
is this being used anywhere? I only see `props.metricName` being used, the other one is Metrics class property.

Reply:
Ya this is used to create the `GitHubWorkflowMonitorAlarms` CDK stack, coming from `infrastructure/lib/infrastructure-stack.ts`